### PR TITLE
Add a test for loading gltf with external textures

### PR DESCRIPTION
**Pull request by luca-della-vedova**
# 🎉 New feature

## Summary

This PR is a followup of https://github.com/gazebosim/gz-common/pull/393 and as discussed in https://github.com/gazebosim/gz-common/pull/393#discussion_r934320258 it adds a test to make sure loading GLTF meshes with external textures works as well.

## Test it

It's a new unit test so it will be handled by CI.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: graphics/src/AssimpLoader_TEST.cc [hunk 1]
```diff
@@ -620,6 +620,30 @@ TEST_F(AssimpLoader, LoadGlTF2Box)
   EXPECT_DOUBLE_EQ(mat->Transparency(), 0.0);
 }
 
+/////////////////////////////////////////////////
+// Open a gltf mesh with an external texture
+TEST_F(AssimpLoader, LoadGlTF2BoxExternalTexture)
+{
+  common::AssimpLoader loader;
+  common::Mesh *mesh = loader.Load(
+      common::testing::TestFile("data/gltf", "PurpleCube.gltf"));
```
Comment (mjcarroll) [inline graphics/src/AssimpLoader_TEST.cc:629]:
```suggestion
      common::testing::TestFile("data", "gltf", "PurpleCube.gltf"));
```

File: graphics/src/AssimpLoader_TEST.cc [hunk 2]
```diff
@@ -620,6 +620,30 @@ TEST_F(AssimpLoader, LoadGlTF2Box)
...
+  EXPECT_STREQ("PurpleCube", mesh->SubMeshByIndex(0).lock()->Name().c_str());
+
+  EXPECT_EQ(mesh->MaterialCount(), 1u);
+
+  const common::MaterialPtr mat = mesh->MaterialByIndex(0u);
+  ASSERT_TRUE(mat.get());
+  // No data (it is a file), but a populated filename
+  EXPECT_EQ(nullptr, mat->TextureData());
+  auto testTextureFile =
+    common::testing::TestFile("data/gltf", "PurpleCube_Diffuse.png");
```
Comment (mjcarroll) [inline graphics/src/AssimpLoader_TEST.cc:643]:
```suggestion
    common::testing::TestFile("data", "gltf", "PurpleCube_Diffuse.png");
```

**Review by mjcarroll — DISMISSED** [review]
Use `joinPaths` rather than explicit slashes.

**Comment by mjcarroll** [comment]
A small nit, but if we are including an image, can it be the gazebo logo?

File: graphics/src/AssimpLoader_TEST.cc (hunk 2 above)
Comment (luca-della-vedova) [reply graphics/src/AssimpLoader_TEST.cc:643]: Done!

**Review by chapulina — APPROVED** [review]

**Pull request merged**